### PR TITLE
MRC-2048: Add dummy endpoints for async calibration

### PR DESCRIPTION
**Pull request by r-ash**
This PR will
* ~~Remove existing calibration endpoint~~ - I've re-added this as it will be easier for hint devs if this doesn't break immediately. Will remove after the async calibration is working, see https://github.com/mrc-ide/hintr/pull/233/commits/ab864fc81110542d5cddba40e68eb03cb8389875
* Add dummy `calibrate/submit` endpoint
* Add dummy `calibrate/status` endpoint
* Add dummy `calibrate/result` endpoint
* Move `model/calibration-options` endpoint to `calibrate/options` for consistency
* Remove `model/result` returning load of json data - this doesn't return any info `model/status` doesn't return (should we just remove it?)
* Did some tidying of validate model options tests, 1 test which was skipped is now enabled, another test which was skipped I have removed
* I removed a regression test for some Zambia data - this data hasn't been kept up to date and has been failing for a while locally (always skipped on CI)

This should then give the hint devs a set contract which they can develop against

Also tagging @EmmaLRussell here to see the format for the API requests & responses.

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
Would it be possible to leave the existing endpoints in place for now, as removing any is going to break HINT?

Yes, agree that the model run result endpoint can come out eventually. This is going to mean changing the logic in the HINT front end a bit, where determingin the complete state of the model fit step is done by checking if we have a non-null model run result.

**Comment by r-ash** [comment]
> Would it be possible to leave the existing endpoints in place for now, as removing any is going to break HINT?
> 
> Yes, agree that the model run result endpoint can come out eventually. This is going to mean changing the logic in the HINT front end a bit, where determingin the complete state of the model fit step is done by checking if we have a non-null model run result.

Is it ok if we leave this sitting on a branch? My plan is to get the whole change implemented on mrc-2040 and then we can merge into master whenever hint side is ready.

**Comment by EmmaLRussell** [comment]
Yeah, that's fine too, as long as @richfitz knows not to merge it! (I guess he does now.)

We will need an error status for both calibrate and run (I think we must already have it for run). 'Empty' model run result on success would mean less work on the HINT side, but probably doesn't really make sense other than that. Since it might be bit tight getting these changes in for the Jan workshops, maybe we could leave the 'empty result' for now while we put in the asynch calibration, and then refactor to remove the empty result separately?
 
> Is it ok if we leave this sitting on a branch? My plan is to get the whole change implemented on mrc-2040 and then we can merge into master whenever hint side is ready.

**Comment by richfitz** [comment]
yeah - there is no PR from mrc-2040 into master, and this PR (and other similar ones Rob is doing) will target that branch)

**Review by richfitz — CHANGES_REQUESTED** [review]
This looks good to me - two very small things here

File: inst/schema/ModelResultResponse.schema.json [hunk 1]
```diff
@@ -2,16 +2,11 @@
...
-      "properties": {
-        "barchart": { "$ref": "BarchartMetadata.schema.json" },
-        "choropleth": { "$ref": "ChoroplethMetadata.schema.json" }
-      },
-      "additionalProperties": false,
-      "required" : [ "barchart", "choropleth" ]
+    "id" : { "$ref": "URI.schema.json" },
+    "complete": {
+      "type": "boolean",
+      "enum": [ true ]
```
[the remark this answers is not in the document]
Comment (r-ash) [reply inst/schema/ModelResultResponse.schema.json:8]: Discussed in PM but I am waiting for front end to tell me what format they want on this.

**Review by richfitz — APPROVED** [review]

**Pull request merged**